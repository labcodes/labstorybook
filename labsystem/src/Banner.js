import React from "react";
import PropTypes from "prop-types";
import { isUndefined } from "lodash";
import Icon from "./Icon";
import TextButton from "./Buttons/TextButton";

export default class Banner extends React.Component {
  static propTypes = {
    /** Text that will be rendered inside the Banner. */
    text: PropTypes.string.isRequired,
    /** */
    type: PropTypes.oneOf(["info", "warn", "error"]),
    /** */
    icon: PropTypes.string.isRequired,
    /** */
    buttonProps: PropTypes.shape({
      text: PropTypes.string,
      onClick: PropTypes.func,
    }),
  };

  static defaultProps = {
    type: "info",
    buttonProps: {
      text: "",
      onClick: undefined,
    },
  };

  icon = () => {
    const { icon } = this.props;
    return icon ? (
      <Icon type={icon} color="white" className="lab-banner__icon" />
    ) : undefined;
  };

  button = () => {
    const { buttonProps, type } = this.props;
    if (buttonProps.text) {
      return type === "warn" ? (
        <TextButton size="normal" skin="dark" text={buttonProps.text} />
      ) : (
        <TextButton size="normal" skin="light" text={buttonProps.text} />
      );
    }
    return null;
  };

  handleClick = (e) => {
    const { buttonProps } = this.props;
    if (!isUndefined(buttonProps.onClick)) {
      buttonProps.onClick(e);
    }
  };

  render() {
    const { text, type } = this.props;
    return (
      <div className={`lab-banner lab-banner--${type}`}>
        {this.icon()}
        <span className="lab-banner__message">{text}</span>
        <span
          className="lab-alert__button"
          onClick={this.handleClick}
          role="presentation"
        >
          {this.button()}
        </span>
      </div>
    );
  }
}
