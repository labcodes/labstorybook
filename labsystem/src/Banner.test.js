import React from "react";
import renderer from "react-test-renderer";

import Banner from "./Banner";

describe("Banner", () => {
  it("renders with base props", () => {
    const renderedComponent = renderer
      .create(<Banner text="Info Banner" type="info" icon="calendar" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with button", () => {
    const renderedComponent = renderer
      .create(
        <Banner
          text="Warn Banner with button"
          type="warn"
          icon="coin"
          buttonProps={{
            text: "Warn Button",
            onClick: () => {},
          }}
        />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
