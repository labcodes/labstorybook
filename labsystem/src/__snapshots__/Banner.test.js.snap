// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders with base props 1`] = `
<div
  className="lab-banner__info"
>
  <span
    className="lab-icon lab-icon--calendar lab-icon--white lab-banner__icon"
  />
  <span
    className="lab-banner__message"
  >
    Info Banner
  </span>
  <span
    className="lab-alert__button"
    onClick={[Function]}
    role="presentation"
  />
</div>
`;

exports[`Banner renders with button 1`] = `
<div
  className="lab-banner__warn"
>
  <span
    className="lab-icon lab-icon--coin lab-icon--white lab-banner__icon"
  />
  <span
    className="lab-banner__message"
  >
    Warn Banner with button
  </span>
  <span
    className="lab-alert__button"
    onClick={[Function]}
    role="presentation"
  >
    <button
      className="lab-btn lab-btn--text lab-btn--normal lab-btn--dark"
      onClick={[Function]}
      type="button"
    >
      
      Warn Button
    </button>
  </span>
</div>
`;
